src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -129,6 +129,11 @@
                     state -> !CONFIG_PROVIDER.getOrCreate(state).getUserTimestampsEnabled(),
                     "User provided timestamps (USING TIMESTAMP)");
 
+    public static final DisableFlag groupByEnabled =
+    new DisableFlag("group_by",
+                    state -> !CONFIG_PROVIDER.getOrCreate(state).getGroupByEnabled(),
```

Comment:
outside the scope of this issue:

Rather than `DisableFlag` maybe we make it `EnabledFlag` and undo the `!` prefix every usage uses...